src/installer.ts

```diff
@@ -164,6 +164,62 @@ async function resolveVersionFromManifest(
   }
 }
 
+// for github hosted windows runner handle latency of OS drive
+// by avoiding write operations to C:
+async function cacheWindowsDir(
+  extPath: string,
+  tool: string,
+  version: string,
+  arch: string
+): Promise<string | false> {
+  if (os.platform() !== 'win32') return false;
+
+  const isHosted =
+    process.env['RUNNER_ENVIRONMENT'] === 'github-hosted' ||
+    process.env['AGENT_ISSELFHOSTED'] === '0';
```

Comment:
I agree with @dusan-trickovic! In my opinion, the current implementation degrades readability and complicates working with the code.

Reply:
process.env['RUNNER_ENVIRONMENT'] === 'github-hosted' - confirms the action runs in github environment 
process.env['AGENT_ISSELFHOSTED'] === '0'  - confirm the action runs in AzDev environment

either condition means we should care about specific hardware configuration

Reply:
Oops... Looks like Dusan's comment is about other things. I was talking about the function in general. Returning a string and boolean is not a good practice in my opinion.

Reply:
Why not? It can not return any boolean value but only `false` and it is fully compatible with returning string but it lets avoid implicit "string to boolean" conversion when checking the result and does not require additional convention "empty string stands for non-executed operation"